src/dag.rs

```diff
@@ -125,4 +125,240 @@ where
 }
 
 #[cfg(test)]
-mod tests {}
+mod tests {
+    use super::*;
+    use petgraph::graph::DiGraph;
+    use std::collections::HashSet;
```

Comment:
@coderabbitai ` DiGraph` and `HashSet` are already imported at the top of the file can caarried into the test via `use super::*`. Please remove these imports.